doc/sphinx/language/extensions/match.rst

```diff
@@ -290,6 +290,26 @@ This example emphasizes what the printing settings offer.
 
        Print snd.
 
+Convention about wildcard patterns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, there is a warning when a pattern variable does not occur
+in a right-hand side of ``=>`` used as catch-all clause for at least
+two distinct patterns. The warning may be deactivated by using a
+variable name starting with ``_`` or by setting ``Set Warnings
+"-duplicated-catch-all-with-unused-named-variable"``.
+
+Here is an example where the warning is activated.
```

Comment:
```suggestion
Convention about wildcard patterns
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

By default, there is a warning when a pattern variable matches
two distinct patterns and is not used
on the right-hand side of ``=>``. The warning can be deactivated by using a
variable name starting with ``_`` or by setting ``Set Warnings
"-duplicated-catch-all-with-unused-named-variable"``.

Here is an example where the warning is activated.
```

Ah, several thoughts:
- Can we find a better title?  I have no idea from the title what the paragraph is about.
- Would you put the message in a `.. warn::` here so it appears in the index?  Then put some of the explanation below it?
- Can we find a shorter name for `duplicated-catch-all-with-unused-named-variable`?

Reply:
> - Can we find a better title?  I have no idea from the title what the paragraph is about.

What about "Convention about unused pattern-matching variables".

> - Would you put the message in a `.. warn::` here so it appears in the index?  Then put some of the explanation below it?

OK, I added a `.. warn::`` and reformulated the paragraph. A second reading might be useful.

> - Can we find a shorter name for `duplicated-catch-all-with-unused-named-variable`?

We should... "duplicated-unused-pattern-matching-variable" is still long. Otherwise, "unused-pattern-matching-variable"?

Reply:
How about "Reporting unused pattern-matching variables"?